runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1920,18 +1917,22 @@ Java_java_lang_Class_getNestMembersImpl(JNIEnv *env, jobject recv)
 			PUSH_OBJECT_IN_SPECIAL_FRAME(currentThread, resultObject);
 			J9Class *nestMember = vmFuncs->internalFindClassUTF8(currentThread, J9UTF8_DATA(nestMemberName), J9UTF8_LENGTH(nestMemberName), classLoader, J9_FINDCLASS_FLAG_THROW_ON_FAIL);
 			resultObject = POP_OBJECT_IN_SPECIAL_FRAME(currentThread);
-
+			J9Class *memberNestHost = NULL;
 			if (NULL == nestMember) {
 				/* If internalFindClassUTF8 fails to find the nest member, it sets
 				 * a NoClassDefFoundError
 				 */
 				goto _done;
-			} else if (NULL == nestMember->nestHost) {
-				if (J9_VISIBILITY_ALLOWED != vmFuncs->loadAndVerifyNestHost(currentThread, nestMember, 0)) {
-					goto _done;
+			} else {
+				nestMember = VM_VMHelpers::currentClass(nestMember);
+				memberNestHost = nestMember->nestHost;
+				if (NULL == memberNestHost) {
+					if (J9_VISIBILITY_ALLOWED != vmFuncs->loadAndVerifyNestHost(currentThread, nestMember, 0, &memberNestHost)) {
+						goto _done;
+					}
 				}
 			}
```

Comment:
I think this would read better if you remove `else`:
```c
			nestMember = VM_VMHelpers::currentClass(nestMember);
			J9Class *memberNestHost = nestMember->nestHost;
			if (NULL == memberNestHost) {
				if (J9_VISIBILITY_ALLOWED != vmFuncs->loadAndVerifyNestHost(currentThread, nestMember, 0, &memberNestHost)) {
					goto _done;
				}
			}
```